packages/semcom-demo-app/src/demo.machine.ts

```diff
@@ -0,0 +1,231 @@
+import { ComponentMetadata } from '@digita-ai/semcom-core';
+import { Session } from '@digita-ai/ui-transfer-components';
+import { Observable, of, forkJoin } from 'rxjs';
+import { map, mergeMap } from 'rxjs/operators';
+import { EventObject, MachineConfig, assign, StateSchema } from 'xstate';
+import { SemComService } from './services/semcom.service';
+
+/* CONTEXT */
+
+export interface DemoContext {
+  components?: ComponentMetadata[];
+  session?: Session;
+  semComService: SemComService;
+  shapeIds?: string[];
+  tags?: string[];
+}
+
+export interface WithSession extends DemoContext {
+  session: Session;
+}
+export interface WithShapeIds extends DemoContext {
+  shapeIds: string[];
+}
+export interface WithComponents extends DemoContext {
+  components: ComponentMetadata[];
+}
+
+/* STATES */
+
+export enum DemoStates {
+  AUTHENTICATING = '[DemoState: Authenticating]',
+  DETECTING_SHAPES = '[DemoState: Detecting shapes]',
+  QUERYING_METADATA = '[DemoState: Querying metadata]',
+  REGISTER_COMPONENTS = '[DemoState: Register components]',
+  IDLE = '[DemoState: Idle]',
+}
+
+export type DemoState =
```

Comment:
What with the rest of the states? Each state should have a context specification here, even if they are just all added to the first option.